lib/cwrap.c

```diff
@@ -0,0 +1,245 @@
+/*
+ * cwrap.c
+ *
+ *  Created on: Oct 17, 2016
+ *      Author: ujay
+ */
+#include <limits.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <errno.h>
+#include <string.h>
+#include"../include/cwrapHeader.h"
+#include <limits.h>
+#include <sys/types.h>
+#include <dirent.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include<assert.h>
+
+
+static Map* map;
+
+Map *initializeMap(int capacity){
+	int i;
+	Map *map=(Map*)malloc(sizeof(Map));
+
+	map->opened_files=(opened_dir_struct*)malloc(capacity*sizeof(opened_dir_struct));
+	assert(map->opened_files!=NULL);
+	map->capacity=capacity;
+	map->length=0;
+	return map;
+
+}
+
+Map* getMap(){
+	if(map==0){
+		map=initializeMap(4);
+	}
+	return map;
+}
+//split file from path
+char* split_path_file(char *relative_path) {
+    const char slash='/';
+   	char *filename;
+	char *dirName;
+	filename= strrchr(relative_path, slash);
+	dirName=strndup(relative_path,strlen(relative_path)- strlen(filename));
+	return dirName;
+}
+//check the capacity of map
+int checkCapacity(){
+	if(map->length>=map->capacity)
+		return -1;
+	else
+		return 0;
+}
+// increases the capacity of map
+Map* increaseMapCapacity(){
+	int i;opened_dir_struct *new_opened_files;
+	new_opened_files=(opened_dir_struct*)malloc((2*map->capacity)*sizeof(opened_dir_struct));
+	assert(new_opened_files!=NULL);
```

Comment:
This assertion should become a recoverable error. Libraries should basically only use assertions when state pre-/post-conditions or invariants have been violated. Even then, there's an argument for using error return values instead of assertions.